project-api-tests/src/test/java/org/ballerina/projectapi/MavenCustomRepoTest.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://wso2.com).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ballerina.projectapi;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+
+import static org.ballerina.projectapi.MavenCustomRepoTestUtils.createSettingToml;
+import static org.ballerina.projectapi.MavenCustomRepoTestUtils.deleteArtifacts;
+import static org.ballerina.projectapi.MavenCustomRepoTestUtils.deleteFiles;
+import static org.ballerina.projectapi.MavenCustomRepoTestUtils.getEnvVariables;
+import static org.ballerina.projectapi.MavenCustomRepoTestUtils.getString;
+import static org.ballerina.projectapi.TestUtils.DISTRIBUTION_FILE_NAME;
+import static org.ballerina.projectapi.TestUtils.OUTPUT_CONTAIN_ERRORS;
+import static org.ballerina.projectapi.TestUtils.executeBuildCommand;
+import static org.ballerina.projectapi.TestUtils.executePullCommand;
+import static org.ballerina.projectapi.TestUtils.executePushCommand;
+
+/**
+ * Tests related to Maven repositories.
+ */
+public class MavenCustomRepoTest {
+
+    private static final String org = "luheerathan";
```

Comment:
Similar to other project api tests, can we use the `bctestorg` in this case as well? It would mean that all test resources reside in the same org, which makes it easier to manage.

Reply:
+1